packages/analysis-utils/src/browserManagement/index.ts

```diff
@@ -105,50 +109,186 @@ export class BrowserManagement {
       headless: this.isHeadless,
       args,
     });
+
     this.debugLog('Browser initialized');
   }
 
+  async clickOnButtonUsingCMPSelectors(page: Page): Promise<boolean> {
+    let clickedOnButton = false;
+
+    try {
+      await Promise.all(
+        CMP_SELECTORS.map(async (selector) => {
+          const buttonToClick = await page.$(selector);
+          if (buttonToClick) {
+            await buttonToClick.click();
+            clickedOnButton = true;
+          }
+        })
+      );
+      return clickedOnButton;
+    } catch (error) {
+      return clickedOnButton;
+    }
+  }
+
+  async clickOnGDPRUsingTextSelectors(
+    page: Page,
+    textSelectors: string[]
+  ): Promise<boolean> {
+    if (textSelectors.length === 0) {
+      return false;
+    }
+
+    try {
+      const result = await page.evaluate((args: string[]) => {
+        let clickedOnButton = false;
+
+        const bannerNodes: Element[] = Array.from(
+          (document.querySelector('body')?.childNodes || []) as Element[]
+        )
+          ?.filter((node: Element) => node && node?.tagName === 'DIV')
+          ?.filter((node) => {
+            if (!node || !node?.textContent) {
+              return false;
+            }
+            const regex =
+              /\b(consent|policy|cookie policy|privacy policy|personalize|preferences|cookies)\b/;
+
+            return regex.test(node.textContent.toLowerCase());
+          });
+
+        bannerNodes?.forEach((node: Element) => {
+          const buttonNodes = Array.from(node?.getElementsByTagName('button'));
+
+          buttonNodes?.forEach((cnode) => {
+            if (!cnode?.textContent) {
+              return;
+            }
+
+            args.forEach((text) => {
+              if (cnode?.textContent?.toLowerCase().includes(text)) {
+                clickedOnButton = true;
+                cnode?.click();
+              }
+            });
+          });
+        });
+
+        return clickedOnButton;
```

Comment:
Will remove the management of clickedOnButton var.
```suggestion
        return bannerNodes?.some((node: Element) => {
          const buttonNodes = Array.from(node?.getElementsByTagName('button'));

          return buttonNodes?.some((cnode) => {
            if (!cnode?.textContent) {
              return false;
            }

            return args.some((text) => {
              if (cnode?.textContent?.toLowerCase().includes(text)) {
                cnode?.click();
                return true;
              }
              
              return false;
            });
          });
        });
```

Reply:
Resolved in latest commits.